compiler/qsc_project/src/project.rs

```diff
@@ -231,8 +234,64 @@ pub trait FileSystemAsync {
         Ok(files)
     }
 
+    async fn collect_sources_from_files_field(
+        &self,
+        project_path: &Path,
+        manifest: &Manifest,
+    ) -> ProjectResult<Vec<PathBuf>> {
+        let mut v = vec![];
+        for file in &manifest.files {
+            v.push(
```

Comment:
just curious -- why this pattern instead of a `.map()`, as you suggested over in the import-package-command PR?

Reply:
Good question, this code is old (I'm just moving it up from line 344 below), but I suspect it was because of the `await` and `?`  . Shoehorning a fallible, async function into `map()` might be possible but just looks awkward I think.